src/coefficients.jl

```diff
@@ -0,0 +1,726 @@
+    data,
+    ref,
+    func::JuMP.GenericQuadExpr,
+    set::MOI.GreaterThan,
+)
+    _get_constraint_data(data, ref, func.aff, set)
+    if !_quadratic_vexity(func, -1)
+        push!(data.nonconvex_rows, ref)
+    end
+    return
+end
+
+function _get_constraint_data(
+    data,
+    ref,
+    func::JuMP.GenericAffExpr,
+    set::MOI.GreaterThan,
+)
+    coefficient = set.lower - func.constant
+    if iszero(coefficient)
+        return
+    end
+    _update_range(data.rhs_range, coefficient)
+    if abs(coefficient) < data.threshold_small
+        push!(data.rhs_small, (ref, coefficient))
+    elseif abs(coefficient) > data.threshold_large
+        push!(data.rhs_large, (ref, coefficient))
+    end
+    return
+end
+
+function _get_constraint_data(
+    data,
+    ref,
+    func::JuMP.GenericQuadExpr,
+    set::Union{MOI.EqualTo,MOI.Interval},
+)
+    _get_constraint_data(data, ref, func.aff, set)
+    push!(data.nonconvex_rows, ref)
+    return
+end
+
+function _get_constraint_data(
+    data,
+    ref,
+    func::JuMP.GenericAffExpr,
+    set::MOI.EqualTo,
+)
+    coefficient = set.value - func.constant
+    if iszero(coefficient)
+        return
+    end
+    _update_range(data.rhs_range, coefficient)
+    if abs(coefficient) < data.threshold_small
+        push!(data.rhs_small, (ref, coefficient))
+    elseif abs(coefficient) > data.threshold_large
+        push!(data.rhs_large, (ref, coefficient))
+    end
+    return
+end
+
+function _get_constraint_data(
+    data,
+    ref,
+    func::JuMP.GenericAffExpr,
+    set::MOI.Interval,
+)
+    coefficient = set.upper - func.constant
+    if !(iszero(coefficient))
+        _update_range(data.rhs_range, coefficient)
+        if abs(coefficient) < data.threshold_small
+            push!(data.rhs_small, (ref, coefficient))
+        elseif abs(coefficient) > data.threshold_large
+            push!(data.rhs_large, (ref, coefficient))
+        end
+    end
+    coefficient = set.lower - func.constant
+    if iszero(coefficient)
+        return
+    end
+    _update_range(data.rhs_range, coefficient)
+    if abs(coefficient) < data.threshold_small
+        push!(data.rhs_small, (ref, coefficient))
+    elseif abs(coefficient) > data.threshold_large
+        push!(data.rhs_large, (ref, coefficient))
+    end
+    return
+end
+
+function _get_constraint_data(data, func::Vector{VariableRef})
+    for var in func
+        push!(data.variables_in_constraints, var)
+    end
+    return
+end
+
+# Default fallback for unsupported constraints.
+_update_range(data, func, set) = false
+
+"""
+    coefficient_analysis(model::Model; threshold_dense_fill_in = 0.10, threshold_dense_entries = 1000, threshold_small = 1e-5, threshold_large = 1e+5)
+
+Analyze the coefficients of a JuMP model.
+
+"""
+function coefficient_analysis(
+    model::JuMP.Model;
+    threshold_dense_fill_in::Float64 = 0.10,
+    threshold_dense_entries::Int = 1000,
+    threshold_small::Float64 = 1e-5,
+    threshold_large::Float64 = 1e+5,
```

Comment:
This is probably too small. Maybe
```suggestion
    threshold_large::Float64 = 1e+7,
```
Things in the millions are pretty common

Reply:
And that is frequently an issue.
So we should tell people to avoid the millions.

See Gurobi:

https://support.gurobi.com/hc/en-us/articles/26284836217361-What-does-Model-contains-large-matrix-coefficient-range-mean

Reply:
I would be happy tightening

Reply:
It'd be helpful to link to the Gurobi docs in the print to the user as a justification for this then?

Reply:
Fair enough, any suggestion on how to check if the link is still online and we are not pointing to nothing? Or just leave it be and wait for users to complain if it goes offline?